tests/data/MerchFactory.ts

```diff
@@ -15,7 +16,21 @@ export class MerchFactory {
   }
 
   public static createOptions(n: number): MerchandiseItemOptionModel[] {
-    return FactoryUtils.create(n, MerchFactory.fakeOption);
+    const options = FactoryUtils.create(n, MerchFactory.fakeOption);
+    if (n === 1) return options;
+
+    const optionMetadata = this.createOptionMetadata(n);
+    return options.map((option, i) => {
+      option.metadata = optionMetadata[i];
+      return option;
+    });
```

Comment:
Instead of creating a matching array, we can add metadata to each option.
```suggestion
    const type = faker.datatype.hexaDecimal(10);
    return options.map((o) => {
      o.metadata = MerchFactory.optionMetadataWith({ type });
      return o;
    });
```